gen/wrap_gdal.jl

```diff
@@ -15,17 +16,113 @@ headerfiles = ["cpl_minixml.h",
 
 headerpaths = [joinpath(gdalpath, h) for h in headerfiles]
 
-# indicates if/not the pair should be wrapped
+skip_expr = [:(const CPL_LSBPTR16 = x),
+             :(const CPL_LSBPTR32 = x),
+             :(const CPL_LSBPTR64 = x),
+             :(const CPLAssert = expr)]
+
+skip_func = [:CPLErrorV] # problem with va_list ihnorton/Clang.jl#17
+
+# all these are typealias of Ptr{Void} and are returned by a wrapped function
+check_nullpointer = [
+    :FilterFunc4ValuesType,
+    :FilterFuncType,
+    :GDALAsyncReaderH,
+    :GDALColorTableH,
+    :GDALContourGeneratorH,
+    :GDALDatasetH,
+    :GDALDriverH,
+    :GDALRasterAttributeTableH,
+    :GDALRasterBandH,
+    :GDALWarpOperationH,
+    :OGRCoordinateTransformationH,
+    :OGRDataSourceH,
+    :OGRFeatureDefnH,
+    :OGRFeatureH,
+    :OGRFieldDefnH,
+    :OGRGeometryH,
+    :OGRGeomFieldDefnH,
+    :OGRLayerH,
+    :OGRSFDriverH,
+    :OGRSpatialReferenceH,
+    :OGRStyleMgrH,
+    :OGRStyleTableH,
+    :OGRStyleToolH
+]
+
+"indicates if the pair should be wrapped"
 function header_wrapped(headerfile, cursorname)
     # don't wrap included files
     headerfile == cursorname
 end
 
+"rewrite certain expressions"
+function rewriter(ex::Expr)
+    # comment out predefined expressions
+    if in(ex, skip_expr)
+        return string("# ", e)
+    end
+
+    # if it is a function
+    if isa(ex, Expr) && ex.head == :function
```

Comment:
I think the check `isa(ex, Expr)` is redundant because we have it in the function signature.